files_test.go

```diff
@@ -282,3 +284,264 @@ func TestUploadFileV2(t *testing.T) {
 		t.Errorf("Unexpected error: %s", err)
 	}
 }
+
+type mockGetUploadURLExternalHttpClient struct {
+	ResponseStatus int
+	ResponseBody   []byte
+}
+
+func (m *mockGetUploadURLExternalHttpClient) Do(req *http.Request) (*http.Response, error) {
+	if req.URL.Path != "files.getUploadURLExternal" {
+		return nil, fmt.Errorf("invalid path: %s", req.URL.Path)
+	}
+
+	return &http.Response{
+		StatusCode: m.ResponseStatus,
+		Body:       io.NopCloser(bytes.NewBuffer(m.ResponseBody)),
+	}, nil
+}
```

Comment:
I considered following the existing once.Do(startServer) approach for testing, but found that simply mocking Client.httpclient would be sufficient. Therefore, I decided to mock it instead.